clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -90,10 +90,6 @@ public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
       mUfsPath = fileInfo.getUfsPath();
       String fileName = PathUtils.temporaryFileName(fileId, mNonce, mUfsPath);
       UnderFileSystem ufs = UnderFileSystem.get(fileName, ClientContext.getConf());
-      String parentPath = (new TachyonURI(mUfsPath)).getParent().getPath();
-      if (!ufs.exists(parentPath) && !ufs.mkdirs(parentPath, true)) {
```

Comment:
Because as of this PR, the path is guaranteed to be created on the master-side when the underlying file is created.